File: cpp/1196_HowManyApplesCanYouPutIntoTheBasket.cpp
public:
    int maxNumberOfApples(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [](const int& a, const int& b) {
            return a < b;
        });
        
        int total = 5000, count = 0;
        for (int apple : arr) {
            if (total - apple < 0) {
                break;
            } else {
                total -= apple;
                ++count;
            }
        }
        return count;
    }
};

File: java/1196_HowManyApplesCanYouPutIntoTheBasket.java
class Solution {
    public int maxNumberOfApples(int[] arr) {
        Arrays.sort(arr);
        int total = 5000; int count = 0;
        for (int apple : arr) {
            if (total - apple < 0) {
                break;
            } else {
                total -= apple; ++count;
            }
        }
        return count;
    }
}
